tests/lib/rules/callback-return.js

```diff
@@ -85,6 +86,40 @@ ruleTester.run("callback-return", rule, {
             options: [["cb", "next"]]
         },
 
+        // allow object methods (https://github.com/eslint/eslint/issues/4711)
+        {
+            code: "function a(err) { if (err) { callback(err); } }",
+            options: [["obj.method"]]
+        },
+        {
+            code: "function a(err) { if (err) { return obj.method(err); } }",
+            options: [["obj.method"]]
+        },
+        {
+            code: "function a(err) { if (err) { obj.method(err); } }",
+            options: [["obj().method"]]
+        },
+        {
+            code: "function a(err) { if (err) { //comment\nreturn obj().method(err); } }",
+            options: [["obj().method"]]
+        },
```

Comment:
Hmm. I'm not sure we should allow parens in this option. That's a bit further down the rabbit hole than we've gone before, and I'm not sure it's necessary.